osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -274,7 +274,13 @@ await Task.Run(async () =>
                     foreach (var filter in Filters)
                     {
                         log($"Performing {filter.GetType().ReadableName()}");
-                        items = await filter.Run(items, cts.Token).ConfigureAwait(false);
+                        var filteredItems = await filter.Run(items, cts.Token).ConfigureAwait(false);
+
+                        // To avoid shooting ourselves in the foot, ensure that we manifest a list after each filter.
+                        //
+                        // A future improvement may be passing a reference list through each filter rather than copying each time,
+                        // but this is the safest approach.
+                        items = filteredItems as List<CarouselItem> ?? filteredItems.ToList();
```

Comment:
Any reason to be playing these cast games rather than change `ICarouselFilter.Run()` to return `Task<List<CarouselItem>>`?

Reply:
I felt like there was an explicit reason I didn't do this (maybe just the input and output being different feels weird?), but let me give it a shot